mqtt_client.c

```diff
@@ -878,6 +882,18 @@ esp_err_t esp_mqtt_client_start(esp_mqtt_client_handle_t client)
     return ESP_OK;
 }
 
+esp_err_t esp_mqtt_client_reconnect(esp_mqtt_client_handle_t client)
+{
+    ESP_LOGI(TAG, "Client force reconnect requested");
+    if (client->state != MQTT_STATE_WAIT_TIMEOUT) {
+        ESP_LOGD(TAG, "Client not waiting for reconnect. Ignore request");
+        return ESP_OK;
```

Comment:
I think it may return `ESP_FAIL` useful for the application needs to know the execution results

Reply:
Agree. Done.